cmake/modules/platform/toolcfg/xlc.cmake

```diff
@@ -19,32 +19,53 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 ###############################################################################
 
-set(OMR_WARNING_AS_ERROR_FLAG -qhalt=w)
+if(OMR_HOST_ARCH STREQUAL "ppc")
+	set(OMR_WARNING_AS_ERROR_FLAG -qhalt=w)
 
-#There is no enhanced warning for XLC right now
-set(OMR_ENHANCED_WARNING_FLAG )
+	# There is no enhanced warning for XLC right now
+	set(OMR_ENHANCED_WARNING_FLAG )
 
-list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
-	-qalias=noansi
-	-qxflag=LTOL:LTOL0
-)
-
-list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -qlanglvl=extended0x)
-
-if(OMR_ENV_DATA64)
 	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
-		-q64
+		-qalias=noansi
+		-qxflag=LTOL:LTOL0
 	)
-else()
-	#-qarch should be there for 32 and 64 C/CXX flags but the C compiler is used for the assembler and it has trouble with some assembly files if it is specified
-	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
-		-q32
-		-qarch=ppc
+
+	list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -qlanglvl=extended0x)
+
+	if(OMR_ENV_DATA64)
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+			-q64
+		)
+	else()
+		# -qarch should be there for 32 and 64 C/CXX flags but the C compiler is used for the assembler and it has trouble with some assembly files if it is specified
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+			-q32
+			-qarch=ppc
+		)
+	endif()
+
+	# Testarossa build variables. Longer term the distinction between TR and the rest
+	# of the OMR code should be heavily reduced. In the mean time, we keep
+	# the distinction
+
+	# TR_COMPILE_OPTIONS are variables appended to CMAKE_{C,CXX}_FLAGS, and so
+	# apply to both C and C++ compilations.
+	list(APPEND TR_COMPILE_OPTIONS
+		-qarch=pwr7
```

Comment:
`-qarch=pwr7` seems wrong for z/OS

Reply:
Similarly to above this is guarded by `OMR_HOST_ARCH STREQUAL "ppc"`